File: 1585-the-kth-factor-of-n/the-kth-factor-of-n.java
class Solution {
    public int kthFactor(int n, int k) 
    {
        ArrayList<Integer> factors = new ArrayList<Integer>();
        for(int i=1; i<=n; i++)
        {
            if(n%i == 0 )
            {
                factors.add(i);
            }
        }
        if(k<=factors.size())
        {
            return factors.get(k-1);
        }
        else
        {
            return -1;
        }
    }
}

File: 1585-the-kth-factor-of-n/the-kth-factor-of-n.cs
class Solution 
{
    public int KthFactor(int n, int k) 
    {
        ArrayList factors = new ArrayList();
        for(int i=1; i<=n; i++)
        {
            if(n%i == 0 )
            {
                factors.Add(i);
            }
        }
        if(k<=factors.Count)
        {
            return (int)factors[k-1];
        }
        else
        {
            return -1;
        }
    }
}